### PR TITLE
WIP: Firmware compression

### PR DESCRIPTION
**Pull request by peterharperuk**
Changes needed to the sdk to support cyw43 firmware compression
Driver changes here https://github.com/georgerobotics/cyw43-driver/pull/44

### CODE REVIEW

**Comment by peterharperuk** [comment]
Marked as WIP as we have to wait for the driver changes to be submitted

File: src/rp2_common/cyw43_driver/CMakeLists.txt [hunk 1]
```diff
@@ -69,13 +90,16 @@ if (EXISTS ${PICO_CYW43_DRIVER_PATH}/${CYW43_DRIVER_TEST_FILE})
...
+    target_include_directories(cyw43_driver_picow INTERFACE
+            $(CYW43_FIRMWARE_LOC) # contains a header file with firmware details
+            )
     target_link_libraries(cyw43_driver_picow INTERFACE
             cyw43_driver_base
             pico_stdlib
             hardware_pio
             hardware_dma
             hardware_exception
+            pico_uzlib
```
Comment (lurch) [inline src/rp2_common/cyw43_driver/CMakeLists.txt:102]: Should `pico_uzlib` be included only if `${CYW43_DECOMPRESS_FIRMWARE}` is set? :shrug:

Comment (peterharperuk) [reply src/rp2_common/cyw43_driver/CMakeLists.txt:102]: I would expect it to have no effect if it's unused?

**Comment by peterharperuk** [comment]
Closing this - replaced with https://github.com/raspberrypi/pico-sdk/pull/1310

**Pull request closed without merging**